### ISSUE TITLE
Understanding CT helpers

### ISSUE THREAD

**Issue by mikenor** [firehol/firehol#403]
Hello, I need some help understanding the CT helpers. It is unclear to me how I am supposed to know what I need for helpers, or what I am supposed to do with such knowledge.

My syslog has this error:

> `nf_conntrack: default automatic helper assignment has been turned off for security reasons and CT-based firewall rule not found. Use the iptables CT target to attach helpers instead.`

I found Issue #241 which suggests this message is triggered because I am using the `all` service in various `interface`s and `router`s.

The documentation for `all` appears to be incomplete. The last sentence is "The following complex services are activated:" but nothing follows.

Issue #241 mentions `helper_all` but I can't find anything about this in documentation. But anyway if I were to set `FIREHOL_CONNTRACK_HELPERS_ASSIGNMENT="manual"` then would that be equivalent to `helper_all=""`? I.e. `helper_all` is not used when assignment method is `manual`? In which case this would make `all` be the exact same as `any`?

According to docs, I cannot use `FIREHOL_CONNTRACK_HELPERS_ASSIGNMENT="firehol"` because it would create rules that aren't NAT-aware. But I don't understand, if I create `cthelper` statements manually, do I need to be doing something special to ensure NAT awareness? I'm unclear on what the issues are with NAT + CT helpers.

For manually creating `cthelper` statements, I do not understand what the documentation means by "trust". What is "trusted traffic"? Do I need to enable "trust" for request packets from clients? For response packets to clients? For request packets to servers? Responses from servers? All of the above? Which side of the firewall I am supposed to put which direction of "trust" on? What about when both sides of the firewall have both clients and servers?

The examples in the `cthelper` documentation don't have adequate explanation of what they are supposed to be examples of. E.g. in the SIP examples, is "we" the SIP phone or the IP-PBX server and who is phoning whom? It would also help if the examples showed the accompanying `interface`/`router` statements that make use of the CT helpers so that the reader can understand what they're for.

Once I create the correct `cthelper` statements, then `all` will automatically use them? Without affecting `any` or `anystateless`?

Basically, I'm lost on CT helpers. Maybe I am overthinking. Mostly firehol is nice because I don't have to worry too much about how the underlying netfilter stuff works, but that seems to fall apart somewhat here. Any guidance much appreciated.
